com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapterContext.java

```diff
@@ -231,4 +232,14 @@ public StepFilters getStepFilters() {
     public IStackFrameManager getStackFrameManager() {
         return stackFrameManager;
     }
+
+    @Override
+    public boolean isDebugMode() {
+        return this.isDebugMode;
+    }
+
+    @Override
+    public void setDebugMode(boolean isDebugMode) {
```

Comment:
Name of this setter is also a little strange for me, any naming suggestion ?

Reply:
Add an enum type `enum LaunchMode { debug, noDebug}`. Then the setter could be `setLaunchMode(LaunchMode mode)`

Reply:
Words from Effective Java:
> Prefer two-element enum types to boolean parameters, unless the meaning of the boolean is clear from the method name. Enums make your code easier to read and to write.